gcc/rust/parse/rust-parse-impl.h

```diff
@@ -4783,6 +4783,16 @@ Parser<ManagedTokenSource>::parse_const_item (AST::Visibility vis,
   // parse constant type (required)
   std::unique_ptr<AST::Type> type = parse_type ();
 
+  // A const with no given expression value
+  if (lexer.peek_token ()->get_id () == SEMICOLON)
+    {
+      lexer.skip_token ();
+      return std::unique_ptr<AST::ConstantItem> (
+	new AST::ConstantItem (std::move (ident), std::move (vis),
+			       std::move (type), std::move (outer_attrs),
+			       locus));
+    }
```

Comment:
I think this is good, but we need to handle it somewhere else, right? e.g. the rest of the AST and HIR are probably expecting that an `impl`'s `const` always has a value, right?

Reply:
yeah, this ICEs:

```rust
struct S;

impl S {
    const D: u8;
}
```

while it should produce a nice error: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=9b8643a863d5350e8b226e8bc9789b88

this should be adressed in a later PR however :)